dandi/utils.py

```diff
@@ -504,3 +509,73 @@ def delayed(*args, **kwargs):
     import joblib
 
     return joblib.delayed(*args, **kwargs)
+
+
+def get_instance(dandi_instance_id):
+    if dandi_instance_id.lower().startswith(("http://", "https://")):
+        redirector_url = dandi_instance_id
+        dandi_id = known_instances_rev.get(redirector_url)
+        if dandi_id is not None:
+            instance = known_instances[dandi_id]
+        else:
+            instance = None
+    else:
+        instance = known_instances[dandi_instance_id]
+        redirector_url = instance.redirector
+        if redirector_url is None:
+            raise ValueError("DANDI instance has no known redirector URL")
+    try:
+        r = requests.get(f"{redirector_url}/server-info")
+        r.raise_for_status()
+    except Exception as e:
+        lgr.warning("Request to %s failed (%s)", redirector_url, str(e))
+        if instance is not None:
+            lgr.warning("Using hard-coded URLs")
+            return instance
+        else:
+            raise RuntimeError(
+                f"Could not retrieve server info from {redirector_url},"
+                " and client does not recognize URL"
+            )
+    server_info = r.json()
+    minversion = Version(server_info["cli-minimal-version"])
```

Comment:
Since we went with simpler `cli-minimal-version` + `cli-bad-versions` (instead of `cli-version` packaging like specification) let's not bring packaging as a dependency.  Our version should be a semver when released, even for release candidates and it should work

```
In [4]: from semantic_version import Version; print(Version('0.6.0-rc1') < Version('0.6.0'))                                                           
True
```
I thought that the tricky part in either case would be the development state version 
```
(git)lena:~/proj/dandi/dandi-cli-master[master]git
$> dandi --version
0.5.0+101.ga115c19
```
but it seems to be perfectly fine!
```
In [1]: from semantic_version import Version; print(Version('0.5.0+101.ga115c19') < Version('0.6.0'))                                                  
True

In [2]: from semantic_version import Version; print(Version('0.5.0+101.ga115c19') < Version('0.5.0'))                                                  
False

In [3]: from semantic_version import Version; print(Version('0.5.0+101.ga115c19') < Version('0.5.1'))                                                  
True
```
since provisioned by https://semver.org/spec/v2.0.0.html#spec-item-10

Reply:
was implemented